src/deluge/gui/menu_item/audio_compressor/compressor_params.h

```diff
@@ -26,21 +27,10 @@ class CompParam final : public UnpatchedParam {
 public:
 	using UnpatchedParam::UnpatchedParam;
 	void readCurrentValue() override {
-		auto value = (int64_t)soundEditor.currentParamManager->getUnpatchedParamSet()->getValue(getP());
-		this->setValue((value * (kMaxMenuValue * 2) + 2147483648) >> 32);
-	}
-	// comp params aren't set up for negative inputs - this is the same as osc pulse width
```

Comment:
Is this accurate? If so the function should probably be named for unipolar param instead of for comp param

Reply:
There's another PR coming that deals with pulse width and does just that :)

Reply:
A bunch of the unipolar params do use the full [-1,1) q31 range but are notionally (e.g. volume). HalfPrecisionUnipolar maybe?

Reply:
I had InternallyUnipolar previously, but HalfPrecisionUnipolar is better!